### PR TITLE
Small update for cache storage

### PR DESCRIPTION
**Pull request by giginet**
I added some features to support remote caches.

- Modify accessibility of CacheKeys
- Modify `CacheStorage` protocol
- Add cache actor indicates uploading/downloading caches or not.

### CODE REVIEW

File: Sources/ScipioKit/Producer/Cache/CacheSystem.swift [hunk 1]
```diff
@@ -86,16 +86,16 @@ extension PinsStore.PinState: Hashable {
 }
 
 public struct CacheKey: Hashable, Codable, Equatable {
-    var targetName: String
-    var pin: PinsStore.PinState
+    public var targetName: String
+    public var pin: PinsStore.PinState
```
Comment (giuk-jung) [inline Sources/ScipioKit/Producer/Cache/CacheSystem.swift:90]: Why did you change to public?

Comment (giginet) [reply Sources/ScipioKit/Producer/Cache/CacheSystem.swift:90]: Because I want to call from external storages.

**Review by giuk-jung — APPROVED** [review]

File: Sources/ScipioKit/Runner.swift [hunk 2]
```diff
@@ -193,10 +193,18 @@ extension Runner {
         }
 
         public enum CacheMode {
+            public enum CacheActor {
+                // Save built product to cacheStorage
+                case producer
+                // Consume stored caches
+                case consumer
+            }
```
Comment (freddi-kit) [inline Sources/ScipioKit/Runner.swift:201]: The name of Actor is confusing, how about `CacheActorKind`?

**Review by freddi-kit — APPROVED** [review]
